wellcomeml/ml/bilstm.py

```diff
@@ -67,27 +88,58 @@ def residual_attention(x1):
             input_length=sequence_length,
             embeddings_initializer=embeddings_initializer,
         )(inp)
+        x = tf.keras.layers.Dropout(
+            self.dropout,
+            noise_shape=(None, sequence_length, 1))(x)
         for _ in range(self.nb_layers):
             x = residual_bilstm(x, l2)
         if self.attention:
             x = residual_attention(x)
-        x = tf.keras.layers.GlobalMaxPooling1D()(x)
-        x = tf.keras.layers.Dense(20, kernel_regularizer=l2)(x)
+        if self.feature_approach == 'max':
+            x = tf.keras.layers.GlobalMaxPooling1D()(x)
+        elif self.feature_approach == 'sum':
+            x = tf.keras.backend.sum(x, axis=1)
+        elif self.feature_approach == 'concat':
+            x = tf.concat(x, axis=1)
+        else:
+            raise NotImplementedError
+        x = tf.keras.layers.Dense(self.dense_size, kernel_regularizer=l2)(x)
         out = tf.keras.layers.Dense(
             nb_outputs, activation=output_activation, kernel_regularizer=l2
         )(x)
-        self.model = tf.keras.Model(inp, out)
+        model = tf.keras.Model(inp, out)
 
         optimizer = tf.keras.optimizers.Adam(lr=self.learning_rate, clipnorm=1.0)
-        metrics = Metrics(validation_data=(X_val, Y_val))
-        self.model.compile(optimizer=optimizer, loss="binary_crossentropy", metrics=[])
+        metrics = [
+            METRIC_DICT[m] if m in METRIC_DICT else m
```

Comment:
What is the rationale behind this? If the user passes another name (say `f1_score`) it will keep the string?

Reply:
The idea is that some metrics can be passed as a string for convenience but others need to be defined explicitly so we support only some metrics via string at the moment and we can expand as we need